Source/ARTBaseMessage.m

```diff
@@ -79,10 +79,18 @@ - (NSInteger)messageSize {
         }
         else if ([self.data isKindOfClass:[NSData class]]) {
             finalResult += [self.data length];
-        } else {
+        }
+        else {
             NSError *error = nil;
+            NSJSONWritingOptions options;
+            if (@available(iOS 13.0, *)) {
```

Comment:
I thought I would load this up quickly in Xcode as I was curious as to whether it was acceptable to just make this runtime switch based on iOS platform detection alone, given Apple's [sparse documentation](https://developer.apple.com/documentation/foundation/nsjsonwritingoptions/nsjsonwritingwithoutescapingslashes?language=objc) on the new enumeration value you are exercising.

But what I then discovered, which perhaps I could have spotted just by looking at the code here, is that you appear to have some scope issues with your local variables. I'm seeing this in Xcode 11.3.1:

![Screenshot 2020-05-18 at 09 41 32](https://user-images.githubusercontent.com/8318344/82192570-1463a580-98ec-11ea-9d78-d3a13ca4758e.png)

Can you take a look, please.

Reply:
@QuintinWillison Nice catch! Thanks. Wondering why I didn't saw the warnings 🤔

Reply:
Now I know why I didn't see those warning: #1020 😄